multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java

```diff
@@ -99,6 +100,27 @@ void testServiceParametersFetchingErrorHandling(boolean isOptionalService) {
         assertSame(((CloudOperationException) exception.getCause()).getStatusCode(), HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = { false, true })
+    void testUpdateServiceKeysWhenDeleteServiceKeysIsSet(boolean serviceExists) {
+        var service = createMockServiceInstance(false);
+        context.setVariable(Variables.SERVICE_TO_PROCESS, service);
+        context.setVariable(Variables.SERVICE_KEYS_TO_CREATE, Map.of());
+        context.setVariable(Variables.DELETE_SERVICE_KEYS, true);
+        if (serviceExists) {
+            Mockito.when(client.getServiceInstance("service", false))
+                   .thenReturn(service);
+        }
+        step.execute(execution);
+        assertStepFinishedSuccessfully();
+        List<ServiceAction> serviceActionsToExecute = context.getVariable(Variables.SERVICE_ACTIONS_TO_EXCECUTE);
+        if (serviceExists) {
+            assertTrue(serviceActionsToExecute.contains(ServiceAction.UPDATE_KEYS), "Actions should contain " + ServiceAction.UPDATE_KEYS);
+            return;
+        }
+        assertFalse(serviceActionsToExecute.contains(ServiceAction.UPDATE_KEYS), "Actions should not contain " + ServiceAction.UPDATE_KEYS);
```

Comment:
Add some empty lines to create a semantic separation: setup - execution - validation.